engine/src/flutter/shell/platform/darwin/common/framework/Source/Logger.swift

```diff
@@ -0,0 +1,133 @@
+  case important
+
+  /// Messages indicating a critical condition. Causes the application to immediately terminate.
+  case fatal
+}
+
+/// A singleton logger for outputting runtime messages.
+///
+/// This logger allows for messages to be logged at different severity levels. Its output can be
+/// filtered by setting the `logLevel` property to the minimum log level to be logged.
+///
+/// **Usage:**
+/// ```swift
+/// Logger.logInfo("Application has started.")
+/// Logger.logLevel = .warning  // Only show warnings and above
+/// Logger.logError("Failed to load asset: \(assetKey)")
+/// ```
+@objc(FlutterLogger) public final class Logger: NSObject {
+  private static var shared = Logger()
+  private let outputWriter: OutputWriter
+  public let logLevel: LogLevel
+
+  public init(outputWriter: OutputWriter, logLevel: LogLevel) {
+    self.outputWriter = outputWriter
+    self.logLevel = logLevel
+  }
+
+  public override convenience init() {
+    #if os(iOS)
+      // On iOS, the user has no access to stdout.
+      // Output can be read from the log by the user or the `flutter` tool.
+      self.init(outputWriter: SyslogOutputWriter(), logLevel: .info)
+    #elseif os(macOS)
+      // On macOS, both the user and the tool read from stdout.
+      self.init(outputWriter: StdoutOutputWriter(), logLevel: .info)
+    #endif
+  }
+
+  public func log(level: LogLevel, _ message: String) {
+    if level.rawValue >= logLevel.rawValue {
+      outputWriter.writeLine(level: level, message)
+    }
+  }
+}
+
+extension Logger {
+  /// Sets the minimum log level.
+  @objc public static var outputWriter: OutputWriter {
+    get { return shared.outputWriter }
+    set(outputWriter) { shared = Logger(outputWriter: outputWriter, logLevel: shared.logLevel) }
+  }
+
+  /// Sets the minimum log level.
+  @objc public static var logLevel: LogLevel {
+    get { return shared.logLevel }
+    set(logLevel) { shared = Logger(outputWriter: shared.outputWriter, logLevel: logLevel) }
+  }
+
+  /// Logs a message at `LogLevel.info`.
+  @objc public static func logInfo(_ message: String) {
+    shared.log(level: .info, message)
+  }
+
+  /// Logs a message at `LogLevel.important`.
+  @objc public static func logImportant(_ message: String) {
+    shared.log(level: .important, message)
+  }
+
+  /// Logs a message at `LogLevel.warning`.
+  @objc public static func logWarning(_ message: String) {
+    shared.log(level: .warning, message)
+  }
+
+  /// Logs a message at `LogLevel.error`.
+  @objc public static func logError(_ message: String) {
+    shared.log(level: .error, message)
+  }
+
+  /// Logs a message at `LogLevel.fatal` and immediately terminates the application.
+  @objc public static func logFatal(_ message: String) {
+    shared.log(level: .fatal, message)
+    abort()
+  }
+
+  /// Logs a message unconditionally.
+  @objc public static func logDirect(_ message: String) {
+    shared.outputWriter.writeLine(level: .important, message)
```

Comment:
maybe document why use `.important`? (i assume it's because it's the highest level besides `fatal`).